src/ledger/LedgerTxn.cpp

```diff
@@ -1891,4 +1893,75 @@ LedgerTxnRoot::encodeHomeDomainsBase64()
 {
     mImpl->encodeHomeDomainsBase64();
 }
+
+void
+LedgerTxnRoot::Impl::prefetch(std::unordered_set<LedgerKey> const& keys)
+{
+    std::unordered_set<LedgerKey> accounts;
+    std::unordered_set<LedgerKey> offers;
+    std::unordered_set<LedgerKey> trustlines;
+    std::unordered_set<LedgerKey> data;
+
+    auto cacheResult =
+        [&](std::unordered_map<LedgerKey, std::shared_ptr<LedgerEntry const>>
+                res) {
+            for (auto const& item : res)
+            {
+                putInEntryCache(item.first, item.second, LoadType::PREFETCH);
```

Comment:
Will this overwrite a LoadType::IMMEDIATE with a LoadType::PREFETCH?

(I doubt it's critical either way, just checking if it's intentional either way)

Reply:
I think this wouldn't happen since `insertIfNotLoaded` checks if a key already exists in the cache.